internal/util/addmetricsannotation/addmetricsannotation.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package addmetricsannotation
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	CNRMMetricsAnnotation        = "cnrm.cloud.google.com/blueprint"
+	DisableKptMetricsEnvVariable = "DISABLE_KPT_METRICS"
+)
+
+// AddMetricsAnnotation adds metrics annotation to track the usage the of kpt
+type AddMetricsAnnotation struct {
+	// Group is the command group e.g., pkg, fn, live
+	Group string
+}
+
+// Process invokes AddMetricsAnnotation kyaml filter on the resources in input packages paths
+func Process(group string, paths ...string) error {
+	for _, path := range paths {
+		inout := &kio.LocalPackageReadWriter{PackagePath: path, PreserveSeqIndent: true, WrapBareSeqNode: true}
+		ama := &AddMetricsAnnotation{Group: group}
+		err := kio.Pipeline{
+			Inputs:  []kio.Reader{inout},
+			Filters: []kio.Filter{kio.FilterAll(ama)},
+			Outputs: []kio.Writer{inout},
+		}.Execute()
+		if err != nil {
+			// this should be a best effort, do not error if this step fails
+			// https://github.com/GoogleContainerTools/kpt/issues/2559
+			return nil
```

Comment:
May be, then never return error from this function ?

Reply:
Agree. Updated.